CHANGELOG.md

```diff
@@ -1,3 +1,65 @@
+0.16.0 (2022-12-006)
+===
+
+Plan
+---
+
+* Refactor `MarkSweep` to work with both our native mark sweep policy and the malloc mark sweep policy backed by malloc libraries.
+
+Allocators
+---
+
+* Add `FreeListAllocator` which is implemented as a `MiMalloc` allocator.
+* Fix a bug in `ImmixAllocator` that alignment is properly taken into consideration when deciding whether to do overflow allocation.
+
+Policies
+---
+
+* Add `MarkSweepSpace`:
+  * It uses our native MiMalloc implementation in MMTk core.
+  * When the feature `malloc_mark_sweep` is enabled, it uses the selected mimalloc library to back up its allocation.
+* Malloc mark sweep now accounts for memory based on page usage, and each malloc library may use a different page size.
+* The immix space now uses the newly added `BlockPageResource`.
+* Changes to Space Function Table (SFT) to improve our boot time and reduce the memory footprint:
+  * Rename the current SFT map to `SFTSparseChunkMap`, and only use it for 32 bits architectures.
+  * Add `SFTSpaceMap` and by default use it for 64 bits architectures.
+  * Add `SFTDenseChunkMap` and use it when we have no control of the virtual address range for a MMTk space on 64 bits architectures.
+
+API
+---
+
+* Add an option `thread_affinity` to set processor affinity for MMTk GC threads.
+* Add `AllocationSemantics::NonMoving` to allocate objects that are known to be non-moving at allocation time.
+* Add `ReferenceGlue::is_referent_cleared` to allow some bindings to use a special value rathen a normal null reference for a cleared referent.
```

Comment:
```suggestion
* Add `ReferenceGlue::is_referent_cleared` to allow some bindings to use a special value rather than a normal null reference for a cleared referent.
```